pkg/kubectl/cmd/util/clientcache.go

```diff
@@ -119,15 +119,18 @@ func (c *ClientCache) clientConfigForVersion(requiredVersion *schema.GroupVersio
 		return copyConfig(c.noVersionConfig), nil
 	}
 
+	// TODO this isn't what we want.  Each clientset should be setting defaults as it sees fit.
+	oldclient.SetKubernetesDefaults(&config)
+	if requiredVersion == nil {
+		requiredVersion = config.GroupVersion
```

Comment:
This builds on the assumption that nil version is the legacy version that was previously enforced always with that old defaults, which are still needed, otherwise we won't have the negotiated serializer set. We should probably fix it in smarter way, but at least this works in both directions.